FreeTubeApp/FreeTube
# Password protect settings

# Title

Add panel to settings page to password protect settings.


## Pull Request Type
- [X ] Feature Implementation

## Related issue
closes #2907 

## Description
Adds a panel to the settings page to password protect all settings.  The password is currently stored in the clear along with other persistent settings.
If no password is set, the panel prompts the user to set a password.  If a password is set, the other settings panels are hidden until the correct password is supplied.  If the password is supplied, the settings are made visible and an option to remove the password is enabled.

## Screenshots 

Dialog to set a password to prevent access to settings
![image](https://user-images.githubusercontent.com/698207/206284506-2ad8e3f9-2f31-4447-bf7a-c26a0b623c92.png)

Dialog to enter a password to access settings (other settings are hidden)
![image](https://user-images.githubusercontent.com/698207/206284598-0c2b3aa5-4135-4cc5-b564-d205c320ac00.png)

Incorrect password prompt
![image](https://user-images.githubusercontent.com/698207/206284677-4d8d49c8-2eb2-4e50-8765-13fc3727955c.png)

Dialog to lock or remove the password when correct password has been entered
![image](https://user-images.githubusercontent.com/698207/206284760-65351c9e-bf36-407d-9d05-8fda4aa76ed6.png)

## Testing

I have tested the PR manually by going through the following steps:
* Start Freetube
* Navigate to settings, observe settings are available
* Open password protect settings panel
* Enter a password, and click set
* Other settings panels are hidden, observe prompt to enter password to unlock
* Enter an incorrect password and click "unlock"; incorrect password dialog show.  Dismiss by clicking ok
* Enter correct password. Observe that settings are unlocked.
* Navigate away from setting, e.g. to subscriptions
* Return to settings, observe they are locked
* Restart app, observe settings are locked
* Enter correct password, click remove password button.  Observe settings are unlocked
* Restart app, observe settings are unlocked

I have not observed any regressions due to my changes to the ft-input component, but would like some advice on what I should look for here.

## Desktop
- **OS:**
Pop_OS
- **OS Version:**
22.04
- **FreeTube version:**
18.0.0
## Additional context
